### ISSUE TITLE
Issues with Location Warp saving waypoints

### ISSUE THREAD

**Issue by Kataiser9** [ReikaKalseki/Reika_Mods_Issues#1411]
Specifically, the ability seems to be operating under very unusual rules regarding when waypoints are saved and when they're reset. I've done some basic testing, and my findings are somewhat hard for me to explain, so I'll just relay the direct results as follows.

Test 1: Client and server (on the same computer) are running a modpack. On the client, I set waypoints. I log out, close/reopen the client, and log in again. Waypoints are gone.

Test 2: Client and server in same setup as above. On the client, I set waypoints. I log out and stop/restart the server, leaving the client open, then rejoin the server. Waypoints are still there.

Test 3: Client and server in same setup as above. On the client, I set waypoints. I log out and open a single-player creative world, and use /chromaprog to unlock everything. I use Location Warp, and my waypoints _from the multiplayer world_ are still there. I make a new waypoint in single-player, then close it and log back into the server; I now see my original waypoints plus the new one from the single-player world.
Continuing from there, I restart the client and rejoin the server. No waypoints. Switch to single-player world - waypoints are there. Switch back to server - waypoints are there again.

Test 4: single-player client on a minimal install. Haven't tested this as thoroughly (and not at all with a client-server setup), but waypoints still persist between worlds.

TLDR, as far as I can tell: waypoint save data is apparently tied to _Minecraft instances_ rather than world/player combinations; the client has no such save data when starting up, but will "remember" it for all worlds upon opening one single-player world.

(I apologize if this is too verbose, but I don't know much about how player data is saved beyond the basics, so this is all kind of incomprehensible to me, and I'd rather err on the side of providing too much information.)

**Comment by OvermindDL1** [comment]
As another data point, same for me, I've not used warping yet at all because of these issues (hence my forgetting to report it, sorry...).

**Comment by ReikaKalseki** [comment]
I have redesigned the ability; it should be better now.
